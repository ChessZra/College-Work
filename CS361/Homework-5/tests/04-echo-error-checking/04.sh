#!/bin/bash

# sequence of 413 & good echo requests

PORT=$@

EOL=$'\r\n'
REQUEST1=$'GET /echo HTTP/1.1'${EOL}$'Header1: '`printf "%1070s" | tr " " "U"`${EOL}${EOL}

REQUEST2=$'GET /echo HTTP/1.1\r
Header1: Value1\r
Header2: Value2\r
Header3: Value3\r\n\r\n'


for _ in $(seq 1 3); do
    printf "$REQUEST1" | nc -N 127.0.0.1 $PORT 2> /dev/null
    printf "\n"
    printf "$REQUEST2" | nc -N 127.0.0.1 $PORT 2> /dev/null
    printf "\n"
done
