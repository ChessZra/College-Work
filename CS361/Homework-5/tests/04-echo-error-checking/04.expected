HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 49

Header1: Value1
Header2: Value2
Header3: Value3
HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 49

Header1: Value1
Header2: Value2
Header3: Value3
HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 49

Header1: Value1
Header2: Value2
Header3: Value3
